hudi-cli/src/main/java/org/apache/hudi/cli/commands/RepairsCommand.java

```diff
@@ -147,14 +149,16 @@ public String overwriteHoodieProperties(
   public void removeCorruptedPendingCleanAction() {
 
     HoodieTableMetaClient client = HoodieCLI.getTableMetaClient();
-    HoodieActiveTimeline activeTimeline = HoodieCLI.getTableMetaClient().getActiveTimeline();
-
-    activeTimeline.filterInflightsAndRequested().getInstants().forEach(instant -> {
+    HoodieTimeline cleanerTimeline = HoodieCLI.getTableMetaClient().getActiveTimeline().getCleanerTimeline();
+    LOG.info("Inspecting pending clean metadata in timeline for corrupted files");
+    cleanerTimeline.filterInflightsAndRequested().getInstants().forEach(instant -> {
       try {
         CleanerUtils.getCleanerPlan(client, instant);
-      } catch (IOException e) {
-        LOG.warn("try to remove corrupted instant file: " + instant);
+      } catch (AvroRuntimeException e) {
```

Comment:
`AvroRuntimeException` will never be catched. `Not an Avro data file` is en `IOException`.

```
// org.apache.avro.file.DataFileReader#openReader


public static <D> FileReader<D> openReader(SeekableInput in,
                                           DatumReader<D> reader)
  throws IOException {
  if (in.length() < MAGIC.length)
    throw new IOException("Not an Avro data file");

  // read magic header
  byte[] magic = new byte[MAGIC.length];
  in.seek(0);
  for (int c = 0; c < magic.length; c = in.read(magic, c, magic.length-c)) {}
  in.seek(0);

  if (Arrays.equals(MAGIC, magic))              // current format
    return new DataFileReader<D>(in, reader);
  if (Arrays.equals(DataFileReader12.MAGIC, magic)) // 1.2 format
    return new DataFileReader12<D>(in, reader);

  throw new IOException("Not an Avro data file");
}
```

Reply:
@lamber-ken : Thanks. I have made changes to specifically look for this message in the exception to detect corruption. Please take a look.

Reply:
👍